tests/phpunit/tests/canonical.php

```diff
@@ -294,4 +294,29 @@ public function test_utf8_query_keys_canonical() {
 
 		delete_option( 'page_on_front' );
 	}
+
+	/**
+	 * Ensure index sitemaps redirects work as expected.
+	 *
+	 * @group sitemaps
+	 * @ticket 50910
+	 */
+	public function test_sitemaps_canonical_redirects() {
+		// Ugly/incorrect versions redirect correctly.
```

Comment:
Fun this passes with `phpunit --filter test_sitemaps_canonical_redirects` but fails with the full run or `phpunit --group canonical`. My suspicion is that either something is hanging around `$wp` from a previous test.

Reply:
see https://core.trac.wordpress.org/ticket/50910#comment:15 for an explanation of why that happens.